From: SeongJae Park <sj@kernel.org>
Date: Mon, 20 Jan 2025 12:58:14 -0800
Subject: [PATCH] mm/damon/paddr: fix set but unused 'last_addr'

The vaiable is set but not used, due to the mistake from copy-and-paste
of the function.  Fix it.

Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202501191712.rj88Gj15-lkp@intel.com/
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index f99bef5abc9d..7f39d96789a3 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -313,14 +313,11 @@ static unsigned int damon_pa_check_accesses(struct damon_ctx *ctx)
 static void __damon_ca_check_access(struct damon_region *r,
 		struct damon_attrs *attrs)
 {
-	static unsigned long last_addr;
 	static unsigned long last_folio_sz = PAGE_SIZE;
-	static bool last_accessed;
-
-	last_accessed = damon_pa_young(r->sampling_addr, &last_folio_sz);
-	damon_update_region_access_rate(r, last_accessed, attrs);
+	static bool accessed;
 
-	last_addr = r->sampling_addr;
+	accessed = damon_pa_young(r->sampling_addr, &last_folio_sz);
+	damon_update_region_access_rate(r, accessed, attrs);
 }
 
 static unsigned int damon_ca_check_accesses(struct damon_ctx *ctx)
-- 
2.39.5

